runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -88,6 +88,7 @@ local function modifiers_from_number(x, modifiers_table)
   return modifiers
 end
 
+--- @async
```

Comment:
What are the implications of this?

Reply:
Any function that has a route to `coroutine.yield()` is marked by the tool as async. The `@async` annotation is make this explicit. This helps find bugs where the code is called from the main thread but acidentally enters a code path that may yield.

I've got gitsigns fully typed with `@async` [here](https://github.com/lewis6991/gitsigns.nvim/pull/1251) and have found it a very useful diagnostic as it finds places where functions that are expected to execute synchronously are calling async functions.